devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusGenerateCode.java

```diff
@@ -53,6 +57,20 @@ public FileCollection getClasspath() {
         return QuarkusGradleUtils.getSourceSet(getProject(), SourceSet.MAIN_SOURCE_SET_NAME).getCompileClasspath();
     }
 
+    @OutputFiles
+    @org.gradle.api.tasks.Optional
+    public Set<File> getGeneratedSourceDirectory() {
+        Set<File> sourcesDirectories = new HashSet<>();
+        Path src = getProject().getProjectDir().toPath().resolve("src");
+        Path mainSrc = src.resolve(SourceSet.MAIN_SOURCE_SET_NAME);
+        Path testSrc = src.resolve(SourceSet.TEST_SOURCE_SET_NAME);
+        for (String codeGenerationSource : CODE_GENERATION_SOURCES) {
+            sourcesDirectories.add(mainSrc.resolve(codeGenerationSource).toFile());
+            sourcesDirectories.add(testSrc.resolve(codeGenerationSource).toFile());
```

Comment:
Would it makes sense to check whether these paths exist before adding them?

Reply:
Or would it be too early at this point?

Reply:
No I think that's doable. I will need to do some refactoring, the generated code is not the default source set location (we have a subfolder per type of file: proto, avdl, ...) and thus, each generated java file needs to be registered to the compileJava task which is not the case when the task is `UP-TO-DATE`..